src/Tooltip/Tooltip.js

```diff
@@ -0,0 +1,141 @@
+import classnames from 'classnames';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+const METHODS_TO_BIND = [
+  'getIdealPosition',
+  'handleMouseEnter',
+  'handleMouseLeave'
+];
+
+class Tooltip extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {isOpen: false};
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  getIdealPosition() {
+    // Get the anchor and position from state if possible. If not, get it from
+    // the props.
+    let anchor = this.state.anchor || this.props.anchor;
+    let position = this.state.position || this.props.position;
+
+    if (this.refs.tooltipContent) {
+      let viewportHeight = global.window.innerHeight;
+      let viewportWidth = global.window.innerWidth;
+      let tooltipContent = ReactDOM.findDOMNode(this.refs.tooltipContent);
+      let tooltipPosition = tooltipContent.getBoundingClientRect();
+
+      // Change the position if the tooltip will be rendered off the screen.
+      if (position === 'right' && tooltipPosition.right > viewportWidth) {
```

Comment:
maybe you can break this method up more by pulling the getting position logic into a `#getPostion` and the getting anchor logic into a `#getAnchor`